### ISSUE TITLE
Gap in change email flow when alias exclusively used for sign-in

### ISSUE THREAD

**Issue by clintfoster** [aws/amazon-cognito-identity-js#515]
We're generating usernames during registration, only requiring users to be aware of their email address when signing in or performing other account management functions, per the following:

>If your application does not require a username, you do not need to ask users to provide one. Your app can create a unique username for users in the background. This is useful if, for example, you want users to register and sign in with an email address and password.

This has worked well, except for a small gap in the change email address flow, as follows:

1. User changes their email address.
2. User waits until the next day to enter the verification code they were sent.
3. By that time, their session has timed out, and they are prompted to sign in. But there is a problem:
  - If they enter their old email address it won't work because it is no longer in effect.
  - If they enter their new email address it doesn't work either because it hasn't been verified.

A possible solution might be to continue honoring the old email address as an alias until the new one has been verified. Effectively the alias change wouldn't take place until the user verifies the new email address. (Arguably that's how it should work anyway.)

This shouldn't be a security issue because the password isn't being changed, only the email address. So if the "wrong" person tries to sign in with the old email address, they couldn't because they would still need to know the password.

**Comment by elasticsearcher** [comment]
@clintfoster IMO the user should also be able to verify their email without having to be logged in, by simply receiving a link they can click and not have to deal with copy-pasting a code.

It would be possible to do this if VerifyUserAttribute didn't require an access token, as then we'd just construct our own link in a CustomMessage lambda trigger with the code baked into the link and the user would be able to confirm their email on any device/browser.

**Comment by elasticsearcher** [comment]
@itrestian Could you take a look at this issue? Are there any plans to let users verify email address changes without having to be logged in?

Security-wise, the user can only change their email when they are logged in. However, why do they need to be logged in to verify their email?

**Comment by yuntuowang** [comment]
It is a known service issue. I will mark this as a feature request. However, we cannot estimate the delivery time at this point.

**Comment by clintfoster** [comment]
@yuntuowang Does marking this as a feature request mean that there is a different system other than GitHub where we can get updates regarding this item? We understand that there is no delivery estimate at this time. But it would be nice to have a way to follow any future progress and see comments from others.

If you re-read the description it seems clear that this is a real problem that will affect the increasingly popular use case of users who sign in with an email alias and are not aware of any username (generated or otherwise). If the user changes their email address and fails to click on the verification code within one day they will be locked out. This would seem to be a common scenario, especially since we've already encountered it with only a small number of users.

**Comment by yuntuowang** [comment]
@clintfoster Yes, probably the fix will be mainly at our service code and we will update on this repo too. There is no other system to track. I totally understand it is a common use case and we will prioritize this case fix.
